src/EnvelopeCreator/EnvelopeCreator.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DocusignBundle\EnvelopeCreator;
+
+use DocuSign\eSign\ApiException;
+use DocusignBundle\EnvelopeBuilder;
+use DocusignBundle\Exception\UnableToSignException;
+use InvalidArgumentException;
+use League\Flysystem\FileNotFoundException;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Routing\Router;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Webmozart\Assert\Assert;
+
+class EnvelopeCreator
+{
+    private $router;
+    private $logger;
+    private $stopwatch;
+    private $createDocument;
+    private $createRecipient;
+    private $defineEnvelope;
+    private $sendEnvelope;
+
+    public function __construct(
+        RouterInterface $router,
+        LoggerInterface $logger,
+        Stopwatch $stopwatch,
+        CreateDocument $createDocument,
+        CreateRecipient $createRecipient,
+        DefineEnvelope $defineEnvelope,
+        SendEnvelope $sendEnvelope
+    ) {
+        $this->router = $router;
+        $this->logger = $logger;
+        $this->stopwatch = $stopwatch;
+        $this->createDocument = $createDocument;
+        $this->createRecipient = $createRecipient;
+        $this->defineEnvelope = $defineEnvelope;
+        $this->sendEnvelope = $sendEnvelope;
+    }
+
+    /**
+     * @throws FileNotFoundException
+     *
+     * @return string path to redirect
+     *
+     * @throw InvalidArgumentException
+     */
+    public function createEnvelope(EnvelopeBuilder $envelopeBuilder): string
+    {
+        try {
+            $this->validate($envelopeBuilder);
+
+            $key = '[Docusign] Create document';
+            $this->stopwatch->start($key);
```

Comment:
This should be moved to a `TraceableEnvelopeCreator` (using decorator design pattern)